Sources/SwiftProtobuf/TextFormatEncoder.swift

```diff
@@ -33,57 +33,91 @@ private let tab = [UInt8](repeating: asciiSpace, count: tabSize)
 
 /// TextFormatEncoder has no public members.
 internal struct TextFormatEncoder {
+    private static let foyerSize = 4096
+    private var foyer = Array<UInt8>(repeating: 0, count: TextFormatEncoder.foyerSize)
```

Comment:
Why not just reserve 4096 bytes worth of capacity directly in `data`?

Reply:
I'm not sure what you mean.  Do you mean starting `data` out with 4096 bytes, or forcing it to extend the allocation by at least 4096 bytes after/before each append?

Reply:
I mean literally adding this to TextFormatEncoder's initializer, and doing nothing else:

```
data.reserveCapacity(4096)
```

Reply:
(The only benefit of this PR that I can see is that it avoids the initial exponential ramp-up from 0 to 4096 bytes in `data`'s backing store. And that is far more easily avoided by simply reserving capacity directly in `data`. I'm pretty sure doing that will be measurably better than the suggested code, too -- e.g., it wouldn't copy data through two buffers for no reason! -- but who knows how good the optimizer is these days...)

Once `data` grows beyond 4096 bytes in size, it doesn't matter if we append 1 byte or 4096, we get the same allocation pattern.

Reply:
I just did that experiment:
* With our current `main` code, this example runs in 44.1s
* With the addition of `data.reserveCapacity`, it takes 43.4s
* With the changes in this PR, it takes 1.5s

I did some profiling of the `main` code.  That showed that over 90% of the total CPU time was being spent in `memmove` triggered by `append(contentsOf:)`.   I'll take another careful look to see if I can narrow down whether that's the `memmove` to reallocate the backing store or the `memmove` that copies data from the string that's being appended.

Reply:
Wow! That is very _very_ unexpected and it definitely needs a closer, stdlib-level look. 

I'd expect copying data from a string to the buffer would be done using `memcpy`, not `memmove` -- unless of course the optimizer is trying to be clever.

I see `putStringValue` is appending data byte by byte (using both `append(_:)` and `append(contentsOf:)`, which is a bit silly -- the stdlib should rather provide an append equivalent of `Array(unsafeUninitializedCapacity:initializingWith:)` to speed that up a bit. However, I really don't see how appending a series of single bytes could ever lead to a drastic increase of Array resizes, _unless there is a critical logic bug within the stdlib_. My own ad-hoc attempts at recreating this did not succeed so far; I'm seeing the expected exponential resizes, not a series of linear ones.